internal/tiger/mcp/db_tools.go

```diff
@@ -157,59 +159,151 @@ func (s *Server) handleDBExecuteQuery(ctx context.Context, req *mcp.CallToolRequ
 	queryCtx, cancel := context.WithTimeout(ctx, timeout)
 	defer cancel()
 
+	// Parse connection string into config
+	connConfig, err := pgx.ParseConfig(details.String())
+	if err != nil {
+		return nil, DBExecuteQueryOutput{}, fmt.Errorf("failed to parse connection string: %w", err)
+	}
+
+	// Choose query execution mode based on whether parameters are present.
+	// Simple protocol supports multi-statement queries but interpolates
+	// parameters client-side (which we don't want to do, for security's sake).
+	// Extended protocol sends parameters separately but doesn't support
+	// multi-statement queries. This means we don't support multi-statement
+	// queries with parameters (pgx will return an error for them when using
+	// QueryExecModeDescribeExec). See [pgx.QueryExecMode] for details.
+	if len(input.Parameters) > 0 {
+		// Use extended protocol to send parameters separately (more secure,
+		// but doesn't support multi-statement queries).
+		connConfig.DefaultQueryExecMode = pgx.QueryExecModeDescribeExec
+	} else {
+		// Use simple protocol to support multi-statement queries when no
+		// parameters are given.
+		connConfig.DefaultQueryExecMode = pgx.QueryExecModeSimpleProtocol
+	}
+
 	// Connect to database
-	conn, err := pgx.Connect(queryCtx, details.String())
+	conn, err := pgx.ConnectConfig(queryCtx, connConfig)
 	if err != nil {
 		return nil, DBExecuteQueryOutput{}, fmt.Errorf("failed to connect to database: %w", err)
 	}
 	defer conn.Close(context.Background())
 
 	// Execute query and measure time
 	startTime := time.Now()
-	rows, err := conn.Query(queryCtx, input.Query, util.ConvertSliceToAny(input.Parameters)...)
-	if err != nil {
-		return nil, DBExecuteQueryOutput{}, fmt.Errorf("query execution failed: %w", err)
+
+	// Queue the query. When using QueryExecModeSimpleProtocol (no parameters),
+	// it's valid to queue a single multi-statement SQL query as the batch.
+	// See the [pgx.Batch.Queue] documentation for details. When using
+	// QueryExecModeDescribeExec (with parameters), queueing a multi-statement
+	// query here will result in an error when executing it below.
+	batch := &pgx.Batch{}
+	batch.Queue(input.Query, util.ConvertSliceToAny(input.Parameters)...)
```

Comment:
We are executing all queries as "batches" now, but that doesn't actually change much beyond letting us iterate through all the result sets when issuing a multi-statement query via the simple query protocol.